pkg/loki/modules.go

```diff
@@ -1829,12 +1829,16 @@ func (t *Loki) initMetastore() (services.Service, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Service methods have tenant auth disabled in the fakeauth.SetupAuthMiddleware call since this is a shared service
 	metastorepb.RegisterMetastoreServiceServer(t.Server.GRPC, m)
 
 	return m, nil
 }
 
 func (t *Loki) initMetastoreClient() (services.Service, error) {
+	if !t.Cfg.IngesterRF1.Enabled && !t.Cfg.QuerierRF1.Enabled {
```

Comment:
Should this be an `||`?

```go
if t.Cfg.IngesterRF1.Enabled || t.Cfg.QuerierRF1.Enabled {
```

Reply:
nope, if either of those flags are set I want to set up the metastore client. If they're both unset I want to exit early